pkgs/development/python-modules/python-lsp-server/default.nix

```diff
@@ -22,8 +22,21 @@
 , setuptools
 , ujson
 , yapf
+, withAutopep8 ? true
+, withFlake8 ? true
+, withMccabe ? true
+, withPycodestyle ? true
+, withPydocstyle ? true
+, withPyflakes ? true
+, withPylint ? true
+, withRope ? true
+, withYapf ? true
```

Comment:
Those are a lot of options. I doubt anyone will test any combination of them. Why do we want them at all? Is there a good reason you want to disable one?

Reply:
[python-lsp-black](https://github.com/python-lsp/python-lsp-black#usage) recommends users to not have `autopep8` and `yapf` installed, there could be other plugins that doesn't work well with the some of these dependencies. The optional dependencies might also add extra lint messages that users might dislike (I don't care about my scripts' complexity, but mccabe is included by default and there is not easy way of removing it).

Reply:
Could we group the options into less logical ones?

Reply:
Not sure if there's a good way to do it

Reply:
hmmm... Then we just do it like this until we have a better idea.